### PR TITLE
Add playwright to show the browse screenshoot on frontend UI

### PR DESCRIPTION
**Pull request by universea**
Add Playwright Integration for Enhanced Web Scraping and Screenshot Capture

Description:
This Pull Request introduces significant improvements to OpenDevin's web scraping capabilities by integrating Playwright. The key motivation behind this change is to replace the existing request-based approach, which is limited to fetching HTML content without rendering JavaScript. Playwright allows us to execute and render JavaScript, enabling a more comprehensive web scraping experience that mirrors actual browser behavior.

Key Changes:
Playwright Integration: Implemented Playwright for web navigation and interaction, enabling dynamic content rendering which was not possible with the request library.
Screenshot Capture: Added functionality to capture and display webpage screenshots in the UI. This feature is particularly useful for visual verification and debugging of scraped web pages.
Code Refactoring: Refactored browse.py to utilize Playwright, enhancing its capabilities to interact with web elements and execute JS code within the context of the page.
UI Updates: Modified Browser.tsx component to display the captured screenshots, improving the overall user experience by providing visual feedback of the scraping results.

bdee6426ee59d761370030ac65c8388

12e57b084a5fb3ab29be5f92a616d35

### CODE REVIEW

**Comment by rbren** [comment]
This is awesome!

Only concern is that, as it is today, we'll start passing the b64 code to the LLMs, which will make its responses worse and more expensive.

To fix that, can you add `if "screenshot" in event: del event['screenshot']` to:
* [This function](https://github.com/OpenDevin/OpenDevin/blob/da69bab11a69f74d3402f3604caf04014fbff62c/agenthub/langchains_agent/langchains_agent.py#L67)
* and [here](https://github.com/OpenDevin/OpenDevin/blob/da69bab11a69f74d3402f3604caf04014fbff62c/agenthub/planner_agent/prompt.py#L142)

**Comment by universea** [comment]
> This is awesome!
> 
> Only concern is that, as it is today, we'll start passing the b64 code to the LLMs, which will make its responses worse and more expensive.
> 
> To fix that, can you add `if "screenshot" in event: del event['screenshot']` to:
> 
> * [This function](https://github.com/OpenDevin/OpenDevin/blob/da69bab11a69f74d3402f3604caf04014fbff62c/agenthub/langchains_agent/langchains_agent.py#L67)
> * and [here](https://github.com/OpenDevin/OpenDevin/blob/da69bab11a69f74d3402f3604caf04014fbff62c/agenthub/planner_agent/prompt.py#L142)

Many thanks! 
I was wondering why my OpenAI API usage ran out today; I've fixed the issue now.

**Comment by sssuperindex** [comment]
I tested it and successfully displayed the browser screenshot.

File: frontend/src/components/Browser.tsx [hunk 1]
```diff
@@ -4,12 +4,30 @@ import { useSelector } from "react-redux";
 import { RootState } from "../store";
 
 function Browser(): JSX.Element {
-  const url = useSelector((state: RootState) => state.browser.url);
+  const { url, screenshotSrc } = useSelector(
+    (state: RootState) => state.browser,
+  );
+  console.log("Brower tsx Screenshot source:", screenshotSrc);
```
Comment (rbren) [inline frontend/src/components/Browser.tsx:10]:
```suggestion
```

**Review by rbren — APPROVED** [review]

**Comment by rbren** [comment]
Looking good! If you fix the conflicts we can get this in

**Comment by universea** [comment]
> Looking good! If you fix the conflicts we can get this in

Sorry, I made a mistake in the operation, can I revert it?

**Pull request closed without merging**